modules/styling/lib/cs.ts

```diff
@@ -215,12 +216,20 @@ type MakeEmotionSafe<T extends string> = T extends `${infer S}label` ? `${S}labe
 type CSSVarName<ID extends string, Name> = `--${ID}-${MakeEmotionSafe<ToString<Name>>}`;
 
 /**
- * For custom themes that do not overwrite every default.
+ * For custom themes that do not overwrite every default. This is the type given to stencil
+ * functions where everything is optional.
  */
 type OptionalVars<T> = {
   [P in keyof T]?: T[P] extends string ? string : OptionalVars<T[P]>; // coerce CSS Variables to a string to allow any overrides in the function
 };
 
+/**
+ * Vars passed to style config functions. The values will always be strings.
+ */
+type RequiredVars<T> = {
+  [P in keyof T]: T[P] extends string ? string : RequiredVars<T[P]>; // coerce CSS Variables to a string to allow any overrides in the function
+};
```

Comment:
This is the same as the `OptionalVars`, but makes everything required (no `undefined`). This is necessary for stencil style functions because the variable always has a value there.